### ISSUE TITLE
Add test_with_transaction to easily test transaction callbacks etc

### ISSUE THREAD

**Issue by dhh** [rails/rails#16831]
TIL that you can actually easily transaction callbacks like after_commit, but it's a bit cumbersome. This is what you have to do today:

```ruby
  uses_transaction :test_completing_todo_will_cause_a_chat_relay
  test 'completing todo will cause a chat relay' do
    recordings(:design_todo).complete
    assert last_relayed_recordable(:anniversary).completed?
  end
```

I'd like to expose this as a more user-friendly and documented feature:

```ruby
  test_with_transaction 'completing todo will cause a chat relay' do
    recordings(:design_todo).complete
    assert last_relayed_recordable(:anniversary).completed?
  end
```

**Comment by dhh** [comment]
@arthurnn Are you still interested in investigating this?

**Comment by arthurnn** [comment]
@dhh i have that code in a patch on Shopify already, will cleanup the code and propose a PR still this weekend.!

**Comment by dhh** [comment]
Was solved in #18458.
